code/game/objects/items/weapons/implants/implant.dm

```diff
@@ -163,63 +170,98 @@ Implant Specifics:<BR>"}
-		var/mob/T = imp_in
-		message_admins("Explosive implant triggered in [T] ([T.key]). (<A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[T.x];Y=[T.y];Z=[T.z]'>JMP</a>) ")
-		log_game("Explosive implant triggered in [T] ([T.key]).")
-		need_gib = 1
-
-		if(ishuman(imp_in))
-			if (elevel == "Localized Limb")
-				if(part) //For some reason, small_boom() didn't work. So have this bit of working copypaste.
-					imp_in.visible_message("<span class='warning'>Something beeps inside [imp_in][part ? "'s [part.name]" : ""]!</span>")
-					playsound(loc, 'sound/items/countdown.ogg', 75, 1, -3)
-					sleep(25)
-					if (istype(part,/obj/item/organ/external/chest) ||	\
-						istype(part,/obj/item/organ/external/groin) ||	\
-						istype(part,/obj/item/organ/external/head))
-						part.createwound(BRUISE, 60)	//mangle them instead
-						explosion(get_turf(imp_in), -1, -1, 2, 3)
-						qdel(src)
-					else
-						explosion(get_turf(imp_in), -1, -1, 2, 3)
-						part.droplimb(0,DROPLIMB_BLUNT)
-						qdel(src)
-			if (elevel == "Destroy Body")
-				explosion(get_turf(T), -1, 0, 1, 6)
-				T.gib()
-			if (elevel == "Full Explosion")
-				explosion_spread(get_turf(T), rand(8,13))
-				T.gib()
-
-		else
-			explosion(get_turf(imp_in), 0, 1, 3, 6)
+	if(!imp_in)
+		small_countdown()
+		return
 
-	if(need_gib)
-		imp_in.gib()
+	message_admins("Explosive implant triggered in [imp_in] ([imp_in.key]). (<A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[imp_in.x];Y=[imp_in.y];Z=[imp_in.z]'>JMP</a>) ")
+	log_game("Explosive implant triggered in [imp_in] ([imp_in.key]).")
+	if(ishuman(imp_in))
+		var/mob/living/carbon/human/T = imp_in
+		if(elevel == "Localized Limb" && part)
+			small_countdown()
+			return	//to avoid hotspot_expose. 
+		if(elevel == "Destroy Body")
+			explosion(get_turf(T), -1, 0, 1, 6)
+			T.gib()
+		if(elevel == "Full Explosion")
+			explosion_spread(get_turf(T), rand(8,13))
+			T.gib()
+	else if(ismob(imp_in))
+		var/mob/M = imp_in
+		M.gib()
+		
+	var/turf/F = get_turf(imp_in)	
+	if(F)
+		F.hotspot_expose(3500,125)
+	qdel(src)
 
-	var/turf/t = get_turf(imp_in)
+/obj/item/implant/explosive/proc/small_countdown()
+	if(!imp_in)
+		visible_message(SPAN_WARNING("Something begins beeping..."))
+	if(ishuman(imp_in))
+		var/message = "Something beeps inside of [imp_in][part ? "'s [part.name]" : ""]..." //for some reason SPAN_X and span() both hate having this in-line
+		imp_in.visible_message(SPAN_WARNING(message))
+	else if(ismob(imp_in))
+		imp_in.visible_message(SPAN_WARNING("Something beeps inside of [imp_in]..."))
+	playsound(loc, 'sound/items/countdown.ogg', 75, 1, -3)
+	addtimer(CALLBACK(src, .proc/small_boom), 25)
 
-	if(t)
-		t.hotspot_expose(3500,125)
+/obj/item/implant/explosive/proc/small_boom()
+	if(!imp_in)
+		explosion(get_turf(src), -1, 0, 2, 4)
+	if(ishuman(imp_in) && part)
+		//No tearing off these parts since it's pretty much killing. Mangle them. 
+		if(istype(part,/obj/item/organ/external/chest) ||	\
```

Comment:
It'd be better to typecast the organ and then check if vital == TRUE.

Reply:
ah, cool, didn't know that was a thing. Time to update oldcode